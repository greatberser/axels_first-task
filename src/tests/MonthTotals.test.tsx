import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

describe('MonthTotals', () => {
  it('should render list of month totals', () => {
    const { asFragment } = render(<div>MonthTotals Component</div>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should display total expenses for the month', () => {
    const mockTotal = 100;
    render(<div>Total Expenses: ${mockTotal.toFixed(2)}</div>);
    expect(screen.getByText(/total expenses: \$100\.00/i)).toBeInTheDocument();
  });

  it('should handle empty total gracefully', () => {
    render(<div>Total Expenses: $0.00</div>);
    expect(screen.getByText(/total expenses: \$0\.00/i)).toBeInTheDocument();
  });
});
