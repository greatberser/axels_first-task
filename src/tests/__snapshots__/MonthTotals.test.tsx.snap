// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`MonthTotals should render list of month totals 1`] = `
<DocumentFragment>
  <div>
    MonthTotals Component
  </div>
</DocumentFragment>
`;
